core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -925,14 +925,26 @@ trait VecLike[T <: Data] extends IndexedSeq[T] with HasId with SourceInfoDoc {
   */
 abstract class Record(private[chisel3] implicit val compileOptions: CompileOptions) extends Aggregate {
 
+  /** Classes can override opaqueType if they want to "unbox"
+    * Records that have maps with a single element. If opaqueType
+    * is overridden to true, the subfield name of the record is
+    * skipped and the Record name is used instead. See RecordSpec
+    * for example usage and expected output
+    */
+  def opaqueType: Boolean = false
+
   // Doing this earlier than onModuleClose allows field names to be available for prefixing the names
   // of hardware created when connecting to one of these elements
   private def setElementRefs(): Unit = {
     // Since elements is a map, it is impossible for two elements to have the same
     // identifier; however, Namespace sanitizes identifiers to make them legal for Firrtl/Verilog
     // which can cause collisions
     val _namespace = Namespace.empty
-    for ((name, elt) <- elements) { elt.setRef(this, _namespace.name(name, leadingDigitOk = true)) }
+    assert(
```

Comment:
```suggestion
    require(
```
`require` is used for indicating the user has made a mistake, as opposed to `assert` which usually means the developer has made a mistake.